integration_test/cases/preload.exs

```diff
@@ -552,6 +552,30 @@ defmodule Ecto.Integration.PreloadTest do
     assert p2.id == c4.post.id
   end
 
+  test "custom preload_order" do
```

Comment:
This is enough for tests? Makes sense to add unit tests for this new option?

Reply:
I think so

Reply:
I think so

Reply:
We generally avoid integration tests. Is it possible for this feature to be tested with unit tests?